code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -914,9 +914,23 @@
 	supports_variations = DIGITIGRADE_VARIATION
 	armor = list(MELEE = 30,  BULLET = 15, LASER = 30, ENERGY = 40, BOMB = 10, BIO = 100, RAD = 50, FIRE = 100, ACID = 100, STAMINA = 60, BLEED = 70)
 	resistance_flags = FIRE_PROOF | ACID_PROOF
+	/// How many charges total the shielding has
+	var/shield_integrity = 60
+	/// How long after we've been shot before we can start recharging.
+	var/recharge_delay = 20 SECONDS
+	/// How quickly the shield recharges each charge once it starts charging
+	var/recharge_rate = 1 SECONDS
+	/// The icon for the shield
+	var/shield_icon = "shield-old"
+
+/obj/item/clothing/suit/space/hardsuit/shielded/Initialize(mapload)
+	. = ..()
+	if(!allowed)
+		allowed = GLOB.advanced_hardsuit_allowed
 
-/obj/item/clothing/suit/space/hardsuit/shielded/setup_shielding()
-	AddComponent(/datum/component/shielded, max_charges = 3, recharge_start_delay = 20 SECONDS, charge_increment_delay = 1 SECONDS, charge_recovery = 1, lose_multiple_charges = FALSE, shield_icon = "shield-old")
+/obj/item/clothing/suit/space/hardsuit/shielded/ComponentInitialize()
```

Comment:
any reason to use componentinit over `setup_shielding` wrapper?

Reply:
oh, no